apache/metrics.go

```diff
@@ -0,0 +1,423 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apache
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
+)
+
+// Status allows for the deserializtion of metrics from an apache server-status endpoint
+type Status struct {
+	ServerVersion                string  `schema:"ServerVersion"`
+	ServerMPM                    string  `schema:"ServerMPM"`
+	ServerBuilt                  string  `schema:"Server_Built"`
+	CurrentTime                  string  `schema:"CurrentTime"`
+	RestartTime                  string  `schema:"RestartTime"`
+	ServerUptime                 string  `schema:"ServerUptime"`
+	Load1                        float64 `schema:"Load1"`
+	Load5                        float64 `schema:"Load5"`
+	Load15                       float64 `schema:"Load15"`
+	CPULoad                      float64 `schema:"CPULoad"`
+	ReqPerSec                    float64 `schema:"ReqPerSec"`
+	BytesPerSec                  float64 `schema:"BytesPerSec"`
+	BytesPerReq                  float64 `schema:"BytesPerReq"`
+	ParentServerConfigGeneration string  `schema:"ParentServerConfigGeneration"`
+	ParentServerMPMGeneration    string  `schema:"ParentServerMPMGeneration"`
+	CPUUser                      int     `schema:"CPUUser"`
+	CPUSystem                    int     `schema:"CPUSystem"`
```

Comment:
There was an issue where if a value didn't exist it would default to zero so I was setting it to null. But the issue was sometimes values would equal zero. I fixed it so that it only converts metrics that don't exist in the server-status?auto will be null